### PR TITLE
Definition Based Extraction: Extract repeating order elements properly.

### LINKED ISSUE #1520: Definition Based Extraction: Issue in extracting repeating order elements
**Describe the bug**
Issue during definition based extraction having repeating order elements, such as **given** in the **Patient.name** resource having a cardinality ratio [0..*]

**Component**
SDC library

**To Reproduce**
1. Use below json as questionnaire string.
```JSON
{
    "resourceType": "Questionnaire",
    "item": [
                {
                  "linkId": "PR-name",
                  "type": "group",
                  "definition": "http://hl7.org/fhir/StructureDefinition/Patient#Patient.name",
                  "item": [
                    {
                      "linkId": "PR-name-text",
                      "definition": "http://hl7.org/fhir/StructureDefinition/Patient#Patient.name.given",
                      "type": "string",
                      "text": "First Name"
                    },
                    {
                      "linkId": "PR-name-middle",
                      "definition": "http://hl7.org/fhir/StructureDefinition/datatypes#Patient.name.given",
                      "type": "string",
                      "text": "Middle Name"
                    }
                  ]
                }
            ]
}
```
2. Perform extraction using the questionnaire response
3. Patient.name element will contain only single value in the given field.

**Expected behavior**
Paitent.name element should contain multiple given values as expected.

@jingtang10

### PR DESCRIPTION
**Pull request by aditya-07**
**IMPORTANT: All PRs must be linked to an issue (except for extremely trivial and straightforward changes).**

Fixes #1520

**Description**
Issue:
* If a group has repeating order elements, last answer overrides the pervious ones during definition based extraction instead of all answers being recorded.

Fix:
* The code was using `set<FieldName>( List<Type>)` api which replaces the old list. Issue is resolved now by using `add<FieldName>(Type)` which appends answer to the list.

**Alternative(s) considered**
No

**Type**
Choose one: Bug fix 

**Screenshots (if applicable)**

**Checklist**
- [x] I have read and acknowledged the [Code of conduct](https://github.com/google/android-fhir/blob/master/docs/CODE_OF_CONDUCT.md).
- [x] I have read the [Contributing](https://github.com/google/android-fhir/wiki/Contributing) page.
- [x] I have signed the Google [Individual CLA](https://cla.developers.google.com/about/google-individual), or I am covered by my company's [Corporate CLA](https://cla.developers.google.com/about/google-corporate ).
- [x] I have discussed my proposed solution with code owners in the linked issue(s) and we have agreed upon the general approach.
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the style guide of this project.
- [x] I have run `./gradlew check` and `./gradlew connectedCheck` to test my changes locally.
- [x] I have built and run the demo app(s) to verify my change fixes the issue and/or does not break the demo app(s).

### CODE REVIEW

**Comment by santosh-pingle** [comment]
@jingtang10 
Could you please review the pr?

File: datacapture/src/main/java/com/google/android/fhir/datacapture/mapping/ResourceMapper.kt [hunk 1]
```diff
@@ -536,6 +535,22 @@ private fun updateFieldWithEnum(base: Base, field: Field, value: Base) {
...
+ * 1. addName() - add an empty HumanName and return it.
+ * 2. addName(HumanName) - adds a new Name to the list.
+ * 3. setName(List<HumanName>) - replaces old list if any with the new list.
+ *
+ * For Parameterized list of primitive type e.g HumanName.given of type List<StringType>
+ * 1. addGivenElement() - add an empty StringType and return it.
+ * 2. addGiven(String) - adds a new StringType to the list.
+ * 3. setGiven(List<StringType>) - replaces old list if any with the new list.
+ */
 private fun updateField(
```
Comment (jingtang10) [inline datacapture/src/main/java/com/google/android/fhir/datacapture/mapping/ResourceMapper.kt:554]: and rename `updateFieldWithAnswer` to `setFieldElementValue`

File: datacapture/src/main/java/com/google/android/fhir/datacapture/mapping/ResourceMapper.kt [hunk 2]
```diff
@@ -536,6 +535,22 @@ private fun updateFieldWithEnum(base: Base, field: Field, value: Base) {
...
+ *
+ * For Parameterized list of custom type e.g Patient.name of type List<HumanName>
+ * 1. addName() - add an empty HumanName and return it.
+ * 2. addName(HumanName) - adds a new Name to the list.
+ * 3. setName(List<HumanName>) - replaces old list if any with the new list.
+ *
+ * For Parameterized list of primitive type e.g HumanName.given of type List<StringType>
+ * 1. addGivenElement() - add an empty StringType and return it.
+ * 2. addGiven(String) - adds a new StringType to the list.
+ * 3. setGiven(List<StringType>) - replaces old list if any with the new list.
```
Comment (jingtang10) [inline datacapture/src/main/java/com/google/android/fhir/datacapture/mapping/ResourceMapper.kt:552]: update this as discussed. describe what APIs are being used in this function, and then list alternative APIs.

File: datacapture/src/main/java/com/google/android/fhir/datacapture/mapping/ResourceMapper.kt (hunk 1 above)
Comment (aditya-07) [reply datacapture/src/main/java/com/google/android/fhir/datacapture/mapping/ResourceMapper.kt:554]: Done

File: datacapture/src/main/java/com/google/android/fhir/datacapture/mapping/ResourceMapper.kt (hunk 2 above)
Comment (aditya-07) [reply datacapture/src/main/java/com/google/android/fhir/datacapture/mapping/ResourceMapper.kt]: Updated the description, didn't add alternatives as they can be easily found from hapi docs.

**Pull request merged**